### PR TITLE
Updated type hints and examples in analysis and utils.

### PR DESCRIPTION
**Pull request by Rudra-23**
I have updated outdated examples and type hints for analysis folder and added types for functions inside utils folder.

### CODE REVIEW

File: ablator/analysis/results.py [hunk 1]
```diff
@@ -39,23 +39,23 @@ def read_result(config_type: type[ConfigBase], json_path: Path) -> pd.DataFrame
...
-    "loss": 0.32
-    }
+    "train_loss": 3.89,
+    "val_loss": 7.04,
+    "current_epoch": 2,
+    }]
     >>> config file
-    experiment_name: "My Experiment"
-    batch_size: 64
+    experiment_dir: "C:\tmp\results\experiment_8925_9991"
```
Comment (fostiropoulos) [inline ablator/analysis/results.py]: Can we switch to unix based path e.g. Ubuntu

**Review by fostiropoulos — CHANGES_REQUESTED** [review]

**Comment by fostiropoulos** [comment]
There are several mypy errors caused by the incorrect use of types. Can you please go over 

https://github.com/fostiropoulos/ablator/blob/v0.0.1-mp/DEVELOPER.md

Additionally, there are many more documentation issues not addressed. When running flake8 they should become obvious. 

@MrLYG can we please add a flake8 test on this branch and resolve all documentation and typing errors.

File: ablator/analysis/results.py (hunk 1 above)
Comment (Rudra-23) [reply ablator/analysis/results.py]: Sure

File: ablator/modules/loggers/main.py
Comment (rudra-guj) [inline ablator/modules/loggers/main.py:1]: For this, the save_checkpoint passes these to torch.save(save_dict, file_path). The save method accepts object and file_path as FILE_LIKE: TypeAlias = Union[str, os.PathLike, BinaryIO, IO[bytes]]. The Path is not included. Therefore, I passed it as string using as_posix(). Please correct me if this is wrong.

Comment (Rudra-23) [inline ablator/modules/loggers/main.py:1]: For this, the save_checkpoint passes these to torch.save(save_dict, file_path). The save method accepts object and file_path as FILE_LIKE: TypeAlias = Union[str, os.PathLike, BinaryIO, IO[bytes]]. The Path is not included. Therefore, I passed it as string using as_posix(). Please correct me if this is wrong.

File: ablator/analysis/results.py
Comment (Rudra-23) [inline ablator/analysis/results.py:1]: For now I have assumed the num_cpus is int type. Please correct me to update.

File: ablator/main/model/wrapper.py [hunk 2]
```diff
@@ -981,6 +1027,21 @@ def make_dataloader_val(self, run_config: RunConfig) -> DataLoader | None:
         -------
         DataLoader | None
             The validation dataloader.
+
+        Examples
+        --------
+        Override the method while using modelWrapper.
```
Comment (fostiropoulos) [inline ablator/main/model/wrapper.py]: `ModelWrapper` ?

File: ablator/main/model/wrapper.py [hunk 3]
```diff
@@ -966,6 +997,21 @@ def make_dataloader_test(self, run_config: RunConfig) -> DataLoader | None:
         -------
         DataLoader | None
             The test dataloader.
+
+        Examples
+        --------
+        Override the method while using modelWrapper.
+
+        class MyModelWrapper(ModelWrapper):
```
Comment (fostiropoulos) [inline ablator/main/model/wrapper.py]: The example does not seem to be properly formatted. Can you please consult @hieuchi911 on how to visualize your changes by running `make docs`

 e.g. 

https://ablator.org/en/latest/ablator.modules.html

https://ablator.org/en/latest/_modules/ablator/modules/optimizer.html#AdamConfig.init_optimizer

File: ablator/modules/loggers/main.py
Comment (fostiropoulos) [reply ablator/modules/loggers/main.py:1]: Is there a page on notion where we document all confusions?

**Review by fostiropoulos — CHANGES_REQUESTED** [review]

**Pull request closed without merging**